File: Тренировка 1.0/02. Линейный поиск/F. Симметричная последовательность/Симметричная последовательность.cs
using System;

short N = short.Parse(Console.ReadLine());
string[] numbers = Console.ReadLine().Split(' ');

for(int i = 0; i < N; i++)
{
    int j = i;
    int k = N - 1;
    while(i <= k && short.Parse(numbers[j]) == short.Parse(numbers[k]))
    {
        j++;
        k--;
    }
    if(j > k)
    {
        Console.WriteLine(i);
        for (int l = i - 1; l >= 0; l--) Console.Write(numbers[l] + " ");
        return;
    }
}

File: Тренировка 1.0/02. Линейный поиск/F. Симметричная последовательность/Симметричная последовательность.cpp
#include <iostream>
#include <vector>

using namespace std;

int main()
{
    short N;
    cin >> N;
    vector<short> numbers(N);
    for (int i = 0; i < N; i++)
    {
        short number;
        cin >> number;
        numbers[i] = number;
    }

    for (int i = 0; i < N; i++)
    {
        int j = i;
        int k = N - 1;
        while (i <= k && numbers[j] == numbers[k])
        {
            j++;
            k--;
        }
        if (j > k)
        {
            cout << i << endl;
            for (int l = i - 1; l >= 0; l--) cout << numbers[l] << " ";
            return 0;
        }
    }
}
